File: Try_GameEngine5/src/com/example/try_gameengine/center_notification/NSANotification.java
package com.example.try_gameengine.center_notification;

import android.content.ContentValues;

/**
 * {@code NSANotification} is a class like NSNotification in iOS.
 * The instances of NSANotification to store all of notification things..
 * @author irons
 *
 */
public class NSANotification {
	/**
	 * name of {@code NSANotification}.
	 */
	private String name;
	/**
	 * object attached to {@code NSANotification}.
	 */
	private Object object;
	/**
	 * useInfo in {@code NSANotification}.
	 */
	private ContentValues userInfo;
	/**
	 * get name of {@code NSANotification}.
	 * @return String.
	 */
	public String getName() {
		return name;
	}
	/**
	 * get object attached to {@code NSANotification}.
	 * @return Object.
	 */
	public Object getObject() {
		return object;
	}
	/** 
	 * get userInfo in {@code NSANotification}.
	 * @return ContentValuse.
	 */
	public ContentValues getUserInfo() {
		return userInfo;
	}
	/**
	 * set name of {@code NSANotification}.
	 * @param name
	 */
	public void setName(String name) {
		this.name = name;
	}
	/**
	 * set object attached to {@code NSANotification}.
	 * @param object
	 */
	public void setObject(Object object) {
		this.object = object;
	}
	/**
	 * set useInfo in {@code NSANotification}.
	 * @param userInfo
	 */
	public void setUserInfo(ContentValues userInfo) {
		this.userInfo = userInfo;
	}
}

File: Try_GameEngine5/src/com/example/try_gameengine/center_notification/NSANotification.kt
package com.example.try_gameengine.center_notification

import android.content.ContentValues

/**
 * `NSANotification` is a class like NSNotification in iOS.
 * The instances of NSANotification to store all of notification things..
 * @author irons
 // */
class NSANotification {
    /**
     * name of `NSANotification`.
     // */
    private var name: String? = null

    /**
     * object attached to `NSANotification`.
     // */
    private var `object`: Any? = null

    /**
     * useInfo in `NSANotification`.
     // */
    private var userInfo: ContentValues? = null

    /**
     * get name of `NSANotification`.
     * @return String.
     // */
    fun getName(): String? {
        return name
    }

    /**
     * get object attached to `NSANotification`.
     * @return Object.
     // */
    fun getObject(): Any? {
        return `object`
    }

    /**
     * get userInfo in `NSANotification`.
     * @return ContentValuse.
     // */
    fun getUserInfo(): ContentValues? {
        return userInfo
    }

    /**
     * set name of `NSANotification`.
     * @param name
     // */
    fun setName(name: String?) {
        this.name = name
    }

    /**
     * set object attached to `NSANotification`.
     * @param object
     // */
    fun setObject(`object`: Any?) {
        this.`object` = `object`
    }

    /**
     * set useInfo in `NSANotification`.
     * @param userInfo
     // */
    fun setUserInfo(userInfo: ContentValues?) {
        this.userInfo = userInfo
    }
}
